### PR TITLE
Liquibase added

### PR DESCRIPTION
**Pull request by gd-kkramarenko**
Added scripts for all tables and loading sample data

### CODE REVIEW

File: EcommerceApp/frontend/angular-ecommerce/README.md [hunk 1]
```diff
@@ -38,3 +39,30 @@ This project was generated with [Angular CLI](https://github.com/angular/angular
 
 Run `ng serve` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.
 
+=======
```
Comment (gd-iistomin) [inline EcommerceApp/frontend/angular-ecommerce/README.md:42]: It looks like git conflict, you should resolve it properly

**Review by gd-iistomin — CHANGES_REQUESTED** [review]

File: EcommerceApp/backend/spring-boot-ecommerce-app/.gitignore [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+build/
+!**/src/main/**/build/
+!**/src/test/**/build/
+
+### VS Code ###
+.vscode/
```
[the remark this answers is not in the document]
Comment (gd-kkramarenko) [reply EcommerceApp/backend/spring-boot-ecommerce-app/.gitignore:33]: Done

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/controller/LoginController.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        this.userService = userService;
+    }
+
+    @GetMapping("/user")
+    public Principal checkUser(Principal user){
+        return user;
+    }
+
+    // endpoint for retrieving user data by username
+    // method is protected by spring security, however
```
[the remark this answers is not in the document]
Comment (gd-kkramarenko) [reply EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/controller/LoginController.java:31]: Done

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/controller/RegistrationController.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    private UserService userService;
+
+    public RegistrationController(UserService userService) {
+        this.userService = userService;
+    }
+
+
+
+    @PostMapping("/users")
+    public ResponseEntity<?> createUser(@RequestBody User user){
```
[the remark this answers is not in the document]
Comment (gd-kkramarenko) [reply EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/controller/RegistrationController.java:23]: Done

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/controller/UserOrdersController.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    public List<Order> getUserOrders(@PathVariable Long customerId){
+        Customer targetCustomer = customerRepository.findCustomerById(customerId);
+
+        List<Order> userOrdersList = new ArrayList<>();
+
+        if(targetCustomer != null){
+            userOrdersList = orderRepository.findOrdersByCustomer(targetCustomer);
+        }
+
+        return  userOrdersList;
```
[the remark this answers is not in the document]
Comment (gd-kkramarenko) [reply EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/controller/UserOrdersController.java:36]: Done

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/dto/UserDetails.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package kkramarenko.ecommerceapp.dto;
+
+import lombok.Data;
+
+@Data
+public class UserDetails {
+
+    // class is used to send back details about user on login to pre-populate forms and bind orders to user.
```
[the remark this answers is not in the document]
Comment (gd-kkramarenko) [reply EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/dto/UserDetails.java:8]: Done

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/entity/User.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+import javax.persistence.*;
+
+@Entity
+@Table(name = "users")
+@Getter
+@Setter
+public class User {
+
+//    FIXME bind users to customers to get list of orders for user
```
[the remark this answers is not in the document]
Comment (gd-kkramarenko) [reply EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/entity/User.java:14]: Done

File: EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/UserServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+     * @return boolean - true on success, false on fail
+     */
+    @Override
+    @Transactional
+    public boolean registerUser(User user) {
+        User userWithMatchingUsername = userRepository.findUserByUsername(user.getUsername());
+        User userWithMatchingEmail = userRepository.findUserByEmail(user.getEmail());
+
+
+        if(userWithMatchingUsername != null || userWithMatchingEmail != null){ return false; }
```
[the remark this answers is not in the document]
Comment (gd-kkramarenko) [reply EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/UserServiceImpl.java:46]: Done

File: EcommerceApp/frontend/angular-ecommerce/README.md (hunk 1 above)
Comment (gd-kkramarenko) [reply EcommerceApp/frontend/angular-ecommerce/README.md:42]: Done

**Review by gd-iistomin — APPROVED** [review]

**Pull request merged**